File: labs/D10_Singleton/python/startingpoint/logger.py
import time

class Logger():

    def __init__(self):
        #Pretend like this is an expensive operation
        time.sleep(3)

    def info(self, message:str):
        print('[info:' + message + ']')

    def warn(self, message:str):
        print('[warn:' + message + ']')

    def error(self, message:str):
        print('[error:' + message + ']')

File: labs/D10_Singleton/java/startingpoint/app/src/main/java/SingletonStartingpoint/Logger.java
package SingletonStartingpoint;

import java.lang.*;

public class Logger {

    public Logger() {
        try {
            // Simulate long init phase.
            Thread.sleep(4000);
        } catch (Exception e) { }
    }
    
    public void warn(String message) {
        System.out.println("[Warn] " + message);
    }

    public void info(String message) {
        System.out.println("[Info] " + message);
    }

    public void error(String message) {
        System.out.println("[Error] " + message);
    }

}
